src/dataviews/dataview-model-base.js

```diff
@@ -98,9 +99,21 @@ module.exports = Model.extend({
     this._setupAnalysisStatusEvents();
   },
 
+  set: function (attributes, options) {
```

Comment:
The "public" way of updating a `DataviewModel` is by using `DataviewModel#update` (https://github.com/CartoDB/cartodb.js/blob/v4/src/dataviews/dataview-model-base.js#L258-L261).  I'd add this logic to that method instead of overriding `set`. Take into account that `update` only accepts an object, so we won't have to support "single attribute updates".

Also, for layers, we've introduced the `setSource` method so that users of the library can change the source more explicitly (https://github.com/CartoDB/cartodb.js/blob/layer-sources/src/geo/map/cartodb-layer.js#L128-L131). We should do something similar here.

Reply:
+1 to add it to `update`.

I did this because in several specs we were modifying `source` via a simple `set`so I realised that's a backdoor, but I agree this can be a bit overkilling.

I'll left untouched the `set`function and will add the logic to `update`.